Sources/SwiftSyntax/SyntaxData.swift

```diff
@@ -356,10 +356,18 @@ struct SyntaxData {
   }
 
   func withLeadingTrivia(_ leadingTrivia: Trivia) -> SyntaxData {
-    return replacingSelf(raw.withLeadingTrivia(leadingTrivia))
+    if let raw = raw.withLeadingTrivia(leadingTrivia) {
+      return replacingSelf(raw)
+    } else {
+      return replacingSelf(raw)
```

Comment:
I think in this case calling `replacingSelf(raw)` is a no-op so we can just `return self` here. Same below.